### DISCUSSION TITLE
Login form password exposed at the client side.

### DISCUSSION THREAD

**Discussion by Sheldenburg** [fastapi/full-stack-fastapi-template#1207] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
def verify_password(plain_password: str, hashed_password: str) -> bool:
    return pwd_context.verify(plain_password, hashed_password)
```

### Description

I have a question regarding the login form. It's probably not a bug, but it sort of bothers me in general... 
At the client side, when we sent the username and password via a form, we are sending password in plain text? I know when we send it over https, it's encrypted on the flight. But if we check the network tab on the browser, we actually exposed the password in the payload (photo attached). 

Any thoughts on this everyone? Would there be some good ways to add extra protection? 
![fastapi-question](https://github.com/tiangolo/full-stack-fastapi-template/assets/18107058/90c8c704-f4df-407f-b664-367f9a1d6e55)

### Operating System

Windows

### Operating System Details

windows 11

### Python Version

Python 3.10.11

**Comment by tiangolo** [comment, marked as answer]
Yep, that's how you would do it with any other system or app. That's why it's very important to use HTTPS. And have in mind that it is only sent from the client to the server, but it's never stored in plaintext in the server nor sent back from the server.
